include/zephyr/sys/util_internal.h

```diff
@@ -136,6 +136,15 @@
 	_51, _52, _53, _54, _55, _56, _57, _58, _59, _60,	\
 	_61, _62, N, ...) N
 
+/* Implementation details for NUM_VA_ARGS */
+#define NUM_VA_ARGS_IMPL(...)                                                                      \
+	COND_CODE_1(                                                                               \
+		IS_EMPTY(__VA_ARGS__), (0),                                                        \
+		(Z_GET_ARG_64(__VA_ARGS__, 63, 62, 61, 60, 59, 58, 57, 56, 55, 54, 53, 52, 51, 50, \
```

Comment:
Question: Couldn't this part be replaced with `UTIL_INC(NUM_VA_ARGS_LESS_1(__VA_ARGS__))`?

Reply:
Didn't try this myself, but the lowest value from `NUM_VA_ARGS_LESS_1` + `UTIL_INC` should be `1` whereas `NUM_VA_ARGS` is `0`

Reply:
The following appears to work:

```c
/* Implementation details for NUM_VA_ARGS */
#define NUM_VA_ARGS_IMPL(...)                                                                      \
	COND_CODE_1(                                                                               \
		IS_EMPTY(__VA_ARGS__), (0),                                                        \
		(UTIL_INC(NUM_VA_ARGS_LESS_1(__VA_ARGS__))))
```

Reply:
@pdgendt thanks for the suggestion, applied the change.
@de-nordic Sorry that I misunderstood your suggestion previously